### PR TITLE
SAK-32398 Remove uses of byte array for uploads

### PR DESCRIPTION
**Pull request by buckett**
When uploading don't read the contents of the upload into memory at all. I've left this as 2 commits because I cleaned up the JavaDoc in the classes I was editing and leaving as 2 commits means it's easier to see what's actually changed without getting confused by the javadoc cleanup.

### CODE REVIEW

**Comment by buckett** [comment]
Previously the streams were not getting closed but now we are explicit about the close and so have to add the `catch` blocks because the close throws the IOException.

File: velocity/tool/src/java/org/sakaiproject/util/ParameterParser.java [hunk 1]
```diff
@@ -232,17 +232,17 @@ public FileItem getFileItem(String name)
...
-            {
-            	return new FileItem(Normalizer.normalize(item.getName(), Normalizer.Form.NFC), item.getContentType(), item.get());
-            }
+			try {
+				return new FileItem(
+						Normalizer.normalize(item.getName(), Normalizer.Form.NFC),
+						item.getContentType(), item.getInputStream()
+				);
+			} catch (IOException e) {
+				// Ignore
```
Comment (ern) [inline velocity/tool/src/java/org/sakaiproject/util/ParameterParser.java]: empty catch, log message.

Comment (buckett) [inline velocity/tool/src/java/org/sakaiproject/util/ParameterParser.java]: I don't like the pattern of logging in a utility class, I see it as the responsibility of the calling class to log if it's `null` and doesn't like that.

I think the better fix it to throw the IOException up to the caller, but I'm not sure it's worth fixing this one like that as using common-upload directly is the better way to go.

**Comment by buckett** [comment]
Added log in `ParameterParser`

**Pull request merged**